### ISSUE TITLE
Finally built an 80 col card but the video is way out of sync and duplicated 4 times what could be the issue?

### ISSUE THREAD

**Issue by Verault** [btb/80ColumnCard#3]
So I finally got around to building one of the 80 col cards. When I do a pr#3 to go into 80 col mode the Sync is way, way off. I need to dial H Hold on my monitor to almost the extreme end to clear up video and as you can see from these photos the video output is duplicated in each corner of the screen. I can see a cursor but when I type it moves but there are no characters. I even tried using the roms from my original Videx card but it still is behaving thr same with those roms.

You can see the cursors on the inverse image. There are 4 cursors. And the non inverse image seems to be bouncing so the cursors ares slightly above each other. I circled them as to make them easier to see:
https://photos.app.goo.gl/85neNsyjyXsYRc43A
and 
https://photos.app.goo.gl/yVLtZP9ghEdJ4nYZ8

A friend of mine is also building thr card. We ordered thr pcbs together. I am waiting on how his card behaves and will post back.

Again what is the issue here and why does the H Hold need to be completely adjusted out to get a video?

**Comment by btb** [comment]
Interesting - can you tell me which firmware and what frequency crystal did you use?

The timings stored in different versions of firmware can vary and different timings may be needed for different monitors but I'm not sure it would result in this kind of problem. Still, you might want to see how it performs on different monitors just to check.

If you have a SoftSwitch hooked up, try it without that.

The sync circuitry is not very complicated, going directly from the 6845 (U19), pins 39 and 40, into the 74LS02 (U2) pins 2 and 3, out pin 1, and through R10, then it's mixed with the pixel data itself and goes into the output transistor Q1.

One possibility is that the 6845 just isn't outputting one of the sync signals - can you try swapping it? Same with the 74LS02 and maybe the 74LS86 (U1).

I'm not much of an expert on the analog side of things, but the "mixing" of the sync with the pixel data is just a very simple voltage adder (averager) that could possibly be sensitive to the exact components used. Are U1 and U2 74LS series chips or something different like 74HCT? Can you verify the values of R9, R10, R11, R12, and R13? Replacing Q1 might be something to try if nothing else seems amiss.

**Comment by Verault** [comment]
Ok. So for the firmware I used the Videx Videoterm ROM 2.4.bin and for the char rom I used the videx-std-7x9-topright.bin and the videx-inverse-topleft.bin with the std 7x9 rom being in the lower part of the eprom and teh videx inverse being in the top portion of the eprom. I used the bin files right from your github download and not outside files.

I wasnt aware the firmware related to the timing so maybe that is it?

I am not using a softswitch.

I am using used 6845 IC's I bought online and have tried several. I even tried the one form my working videx card.. no change. ( I also swapped the used ones I bought on the original videx card and verified they were working.


I tested the 74ls03 and 74ls86 but I can try swapping them for a different brand just because I have plenty.

For reference U1 is a hitachi HD74ls86p and U2 is a generic 74ls02N

R9 is a 1.47K (1.5K) Ohm
r10 2K Ohm
r11 10 Ohm
r12 150 Ohm
r13 218 Ohm (I must have used a 220 ohm instead of a 200 ohm)

The crystal is a 17.430 actual photo of it here: https://photos.app.goo.gl/KRUckTHRnUHQzAbWA

I was wondering if Q1 could be the issue. I can try swapping it for another 2N3904.

Just for reference I am using a Zenith monochrome composite monitor. Its my bench monitor and I used it for everything. I have plenty of other monitors I can try just to say we did that however I  jut want to point out that this zenith monitor outputs from my actual Videx card good 80 col video without having to adjust the H hold at all.

Also Let me ask you this, and please do not take any offence at all by it. Have you built this card or are they more preliminary plans?

**Comment by btb** [comment]
None of that seems amiss. I think that's the exact same crystal I used as well.

I do have a working card that I built and used with a Monitor III and Monitor IIe. You used my gerbers_0.zip file, right?

I just realized I missed in the original post that you have a real videx card and tried swapping the roms - As far as I know, the real Videx cards all used a 2708 EPROM for the firmware and you can't swap to a 2716 or similar without modifying the card. This could potentially have damaged your EPROM?

What eproms and sram chips are you using?
I only recently added this wiki page to clarify some things about the different chip requirements and where the data has to go, did you happen to see it?
https://github.com/btb/80ColumnCard/wiki/Parts

**Comment by Verault** [comment]
The original Roms went back in the videx card and its still working perfectly fine.

The sram is HM6116L-70  12 pin

The eproms I am using are your run of the mill 27C64 which I have lots of. Like I mentioned I used your recommendation on the PARTS wiki and have the videx std 7 x 9 bin in the lower part of the eprom and the videx inverse in the top of the eprom. If there is a way I can upload or share the bin I created I can. As maybe I botched that whole process up.

I have seen your PARTS wiki and have the 6116 in the lowest part of the socket down at the bottom of the board.

The only thing I was unsure about was the cutting of X21 jumper for A11. I didnt think my 6116 qualified for that.

**Comment by btb** [comment]
You should be able to drop files here and I can look at them, yes. If you want to email me directly, it's bradley.bell at gmail.

You shouldn't need to change the jumper for a 6116.

**Comment by Verault** [comment]
I wasnt aware I could add them here.
[Videx80colroms.zip](https://github.com/btb/80ColumnCard/files/11875155/Videx80colroms.zip)
These bins were made to use 27c64 eproms.

**Comment by ryucats** [comment]
I've spent a week trying to get this card working, just managed to do so, and documented it in this pull request: https://github.com/btb/80ColumnCard/pull/4

I suspect that the horizontal timing in the firmware binary you're using is not playing nicely with your monitor; the same thing happened to me.  The timing data in the binary included in the firmware/ subdirectory is not stock; the pull request includes a .s file that can be modified to easily generate a binary with modified timing.

(most of the rest of the parts were scavenged from a faulty Sekom Videx clone, with the exception of the EPROMs, the RAM (I also used a 6116, and a 74ACT273), so I don't think your parts are bad, just that your timing parameters are incompatible with your monitor)

Not included in the pull request is a python script I wrote to derive timing register values from basic parameters -- I didn't include it because my problem turned out to be a) a character generator EPROM that was too fast and b) an incorrect assumption that the timing in the .bin in the firmware/ subdirectory was unmodified.  I can pass that script along if you want to play with it.

**Comment by btb** [comment]
yeah, those files should work. That risk of damage I think would only be if you put your newly-burned ROM into your original Videx but it doesn't sound like you did that. (and a 2764 wouldn't fit of course!)

My apologies for the trouble you're having, I'm really not sure what the problem is. I'm used to just building things for myself, putting them out there for everyone else is kinda new for me. Maybe you could send me your card and I'll get it working for you. That would allow me to see what the pitfalls happen to be so others will avoid it.

@ryucats that's very interesting. Wouldn't they still expect to see some signs of characters even if the timing is wrong?

I don't really know for sure what the "stock" timings should be, I know the one I have up is a little different from the source listing in the manual, but I'm pretty sure Videx themselves were known for tweaking the timings over time.

**Comment by ryucats** [comment]
Just had a look at the timing in the zipfile you attached.  At offset $A1, you have:

`7a 50 5e 2f 22 00 18 1d 00 08 e0 08 00 00 00 00`

Try:

`7b 50 5e 29 1b 08 18 19 00 08 e0 08 00 00 00 00`

... which is per the Videx documentation and fixed my horizontal sync issues.

**Comment by btb** [comment]
Yeah, that's worth a try. That may fix the sync issue, and then a slower character ROM is needed to get the characters to show up...

I used an old 27C32 in mine, and it looks like @ryucats used a M27C64-200.

**Comment by btb** [comment]
Sounds like you may have already tried U20 from your videx card in the new card and it didn't result in visible characters - but that might have been because it's a 2716 and those aren't supported on this card (A11 is needed to switch between standard and alternate character sets, and that line will instead go to VPP on a 2716) - but it is possible that that the Videx character rom chip will work, but only when you tell the card to use the alternate character set.

**Comment by ryucats** [comment]
>Wouldn't they still expect to see some signs of characters even if the timing is wrong?

Independent of timing issues, If @verault is using a too-fast character EPROM, or a 2716 as you noted, then he'd either see garbage or a blank screen (except for the cursor) ... which is exactly what those pictures appear to be.

FWIW, there's a lower bound on the firmware EPROM speed as well.  I thought I had it working with a 100ns chip, but after loading the logic board up with a bunch of other cards, it ended up just beeping when I issued a PR#3.  150ns seems to be okay, but I think 200ns is the safest.

**Comment by ryucats** [comment]
There's a nifty online 6845 timing simulator that you can use to play around and get things in the general ballpark.  It's written for the BBC Micro, and is therefore not going to be a perfect simulation of the Videx clone output, but it'll give you an idea of what your timings are going to do before you burn them to ROM.

Link is [here](https://bbcmic.ro/#%7B%22v%22%3A1%2C%22program%22%3A%22REM%20Intro%20to%20CRTC%20Registers%20by%20Kieran%20Connell.%5CnREM%20CRTC%20Register%20explorer.%5CnREM%20Use%20cursor%20keys%20to%20change%20register%20values.%5CnREM%20Press%20R%20to%20reset%20to%20MODE%201%20defaults.%5CnMODE%201%5CnVDU%2019%2C0%2C4%3B0%3B%5Cn*FX4%2C1%5CnON%20ERROR%20REPORT%3APRINT%3B%5C%22%20at%20line%20%5C%22%3BERL%3AEND%5CnFOR%20Y%25%3D0%20TO%2031%3APRINT%20TAB%280%2CY%25%29%3BY%25%3B%3ANEXT%5CnFOR%20X%25%3D0%20TO%2039%3APRINT%20TAB%28X%25%2C0%29%3BX%25MOD10%3B%3ANEXT%5CnPRINT%20TAB%2810%2C21%29%3B%5C%22Use%20cursor%20keys%20to%5C%22%3BTAB%2810%2C22%29%3B%5C%22change%20register%20values%5C%22%3BTAB%2810%2C23%29%3B%5C%22Press%20R%20to%20reset%5C%22%5CnM%25%3D11%3ADIM%20REGS%28M%25%29%3ADIM%20NAME%24%28M%25%29%5CnPROCresetregs%5CnA%25%3D0%5CnREPEAT%5CnPRINT%20TAB%289%2C17%29%3B%5C%22Total%20scanlines%20%3D%20%5C%22%3B%28REGS%284%29%2B1%29*%28REGS%289%29%2B1%29%2BREGS%285%29%5CnPRINT%20TAB%2811%2C18%29%3B%5C%22Displayed%20RAM%20%3D%20%5C%22%3B%28REGS%281%29*8*REGS%286%29%29%2F1024%3B%5C%22K%20%20%20%20%5C%22%5CnPRINT%20TAB%284%2C4%2BA%25%29%3B%5C%22%3D%3E%5C%22%3B%5CnK%3DGET%5CnPRINT%20TAB%284%2C4%2BA%25%29%3B%5C%22%20%20%5C%22%3B%5CnIF%20K%3D139%20AND%20A%25%3E0%20THEN%20A%25%3DA%25-1%5CnIF%20K%3D138%20AND%20A%25%3CM%25%20THEN%20A%25%3DA%25%2B1%5CnIF%20K%3D136%20THEN%20PROCsetreg%28A%25%2CREGS%28A%25%29-1%29%5CnIF%20K%3D137%20THEN%20PROCsetreg%28A%25%2CREGS%28A%25%29%2B1%29%5CnIF%20K%3DASC%28%5C%22R%5C%22%29%20THEN%20PROCresetregs%5CnUNTIL%20FALSE%5CnEND%5CnDEF%20PROCresetregs%5CnRESTORE%5CnFOR%20I%25%3D0%20TO%20M%25%3AREAD%20NAME%24%28I%25%29%2CREGS%28I%25%29%3APROCsetreg%28I%25%2C%20REGS%28I%25%29%29%3ANEXT%5CnENDPROC%5CnDEF%20PROCprintreg%28R%25%29%5CnPRINT%20TAB%287%2C4%2BR%25%29%3BNAME%24%28R%25%29%3BTAB%2829%2C4%2BR%25%29%3B%5C%22R%5C%22%3BR%25%3B%5C%22%3D%5C%22%3BREGS%28R%25%29%3B%5C%22%20%20%5C%22%5CnENDPROC%5CnDEF%20PROCsetreg%28R%25%2C%20V%25%29%5CnREGS%28R%25%29%3DV%25%5Cn%3F%26FE00%3DR%25%3A%3F%26FE01%3DV%25%5CnPROCprintreg%28R%25%29%5CnENDPROC%5CnREM%20CRTC%20Register%20defaults%20for%20MODE%201.%5CnDATA%20%5C%22Horizontal%20total%20-1%5C%22%2C%20127%20%20%3A%20REM%20R0%5CnDATA%20%5C%22Horizontal%20displayed%5C%22%2C%2080%20%20%3A%20REM%20R1%5CnDATA%20%5C%22Horizontal%20sync%20pos%5C%22%2C%2098%20%20%20%3A%20REM%20R2%5CnDATA%20%5C%22Horizontal%20sync%20width%5C%22%2C%20%2628%3A%20REM%20R3%5CnDATA%20%5C%22Vertical%20total%20-1%5C%22%2C%2038%20%20%20%20%20%3A%20REM%20R4%20%20%20%20%20%20%20%5CnDATA%20%5C%22Vertical%20total%20adjust%5C%22%2C%200%20%20%3A%20REM%20R5%5CnDATA%20%5C%22Vertical%20displayed%5C%22%2C%2032%20%20%20%20%3A%20REM%20R6%5CnDATA%20%5C%22Vertical%20sync%20pos%5C%22%2C%2035%20%20%20%20%20%3A%20REM%20R7%5CnDATA%20%5C%22Interlace%20control%5C%22%2C%200%20%20%20%20%20%20%3A%20REM%20R8%5CnDATA%20%5C%22Scanlines%20per%20row%20-1%5C%22%2C%207%20%20%20%3A%20REM%20R9%5CnDATA%20%5C%22Cursor%20start%5C%22%2C%20%2667%20%20%20%20%20%20%20%20%20%3A%20REM%20R10%5CnDATA%20%5C%22Cursor%20end%5C%22%2C%208%20%20%20%20%20%20%20%20%20%20%20%20%20%3A%20REM%20R11%5Cn%22%7D)... there's a crapload of embedded BASIC in that URL, unfortunately, but I found it works well.

Edit: I plugged @verault 's timings into that, and the effective display is offset several characters to the left.  On a real CRT, that would definitely cause horizontal mirroring.

**Comment by btb** [comment]
I double-checked on my card and it is using the
`7a 50 5e 2f 22 00 18  1d 00 08 e0 08 00 00 00 00`
timings, and it definitely works on my Monitor IIe, FWIW. Doesn't work on the LCD tv I just picked up though.

**Comment by Verault** [comment]
Sorry for not chiming back in. I have been ill. Good to see alot of discussion about the issue. I am going to try the new code in the firmware first. (By the way which rom are we talking about U20 or U3?)

Also I guess the eprom latency makes sense being an issue but this creates a huge problem of its own. I buy alot of eproms and MOST of them work, but have been re-badged. For instance my 27c64 eproms all say ST brand 27c64 but thier Chip ID's come up with a bunch of different manufacturers. So how can I possible find a trustworthy M27C64-200 with 200 latency? Do you possibly have a source I can purchase from thats reputable?

**Comment by btb** [comment]
No problem, I hope you are feeling better.

The firmware with the monitor timings issue is U3, the character generator with has the speed/latency issue is U20.

I like getting "old" chips like these from unicornelectronics.com. $30 minimum order but that's pretty easy to meet if you have a few projects going on. jameco.com has always been very good as well. Both places have always been reliable for me.

**Comment by btb** [comment]
Here's a copy of the firmware for U3, with the "stock" timings from the manual, and ready to go on a 2764
[Videx_Videoterm_ROM_2.4-stocktimings-2764.bin.zip](https://github.com/btb/80ColumnCard/files/11899147/Videx_Videoterm_ROM_2.4-stocktimings-2764.bin.zip)

**Comment by Verault** [comment]
Thank you. And just to confirm the only eprom which is sensitive to latency is u20?

**Comment by btb** [comment]
as far as we know, yes. I currently have a 150ns EEPROM for my U3 and that works fine.

**Comment by Verault** [comment]
Ok @btb and @ryucats I am sorry for the delay in replying but I have more info. I had a friend of mine who is building the same 80 col board mail it to me so I can test his as well. ITs basically behaving the same as mine except his is showing 6 screens (and again no characters) see photo.
![IMG_20230707_120250](https://github.com/btb/80ColumnCard/assets/64697404/4124cc48-d42e-46da-ac60-d4ddfd891ee9)

Im at a complete loss on this. I have MANY MANY eproms of various sizes but like I sai
[27c512bins.zip](https://github.com/btb/80ColumnCard/files/11984219/27c512bins.zip)
d most have been relabeled so the only way to know the actual brand is to read the CHIP ID in my eprom programmer but that wont tell me the latency/speed.  I bought some AMD 27c512-200DC (thats 200ns correct) and I am getting the same exact image,.,, multiple screens inverted, no characters, very much out of sync.

@ryucats I have tried the stock and the timings you recommended. Please see the newest bins I burned , ill attach them here. 

Did I mess up in making them?

I really want these boards to work but I dont know what is wrong at this point. I tried many eproms. I even tried some SST 27SF256 EEPROMS with the same results.

**Comment by btb** [comment]
send me your address and I'll mail you a couple of known-working character roms

**Comment by Verault** [comment]
Thank you. I just emailed you.

**Comment by btb** [comment]
Well we found the problem on Verault's card, and it was due to a "bad" 74LS04 chip.
I have some of the same chips in my collection and they don't work for me either, It's possible these are counterfeit chips being sold as LS when they are actually some other type of logic. This is what they look like:
![image0](https://github.com/btb/80ColumnCard/assets/149551/8f94e7b4-7785-4580-8f7c-f71712663f81)

The 74LS04 is used in the clock circuit, which is another sensitive area. Possibly these chips, or a wider range of logic families, could be made to work with different values of C3 or R1 and R2.

**Comment by Verault** [comment]
BTB is correct. The issue was the 74ls04 chips. They must be some relabeled nonsense from China. They test fine but dont work properly in this card. I ordered new ones from Jameco and they work fine, as the one I received back from BTB. Thank you so much for your help. That was an odd one. Issue resolved.
